user-service/controller/user-controller.js

```diff
@@ -67,28 +67,50 @@ export const deleteUser = async (req, res) => {
 
 export const loginUser = async (req, res) => {
   const { username, password } = req.body
-
   if (!(username && password)) {
     return res
       .status(400)
       .json({ message: 'Username and/or Password are missing!' })
   }
-
   const user = await _findUserByUsername(username)
   if (user?.err) {
     return res.status(500).json({ message: 'ERROR: Could not log user in' })
   }
-
   if (user && (await bcrypt.compare(password, user.password))) {
     const token = jwt.sign({ username, password }, process.env.TOKEN_SECRET, {
       expiresIn: JWT_EXPIRY,
     })
     return res.status(200).json({ userId: user._id, token })
   }
-
   return res.status(401).send('Invalid login credentials')
 }
 
+export const updateUserPassword = async (req, res) => {
+  try {
+    const { username, newPassword } = req.body
+    if (!(username && newPassword)) {
+      return res
+        .status(400)
+        .json({ message: 'Username and/or Password are missing!' })

```

Comment:
Just a note to self here: maybe we can consider implementing a query validator middleware next time to extract the boilerplate logic here that's used in every request handler.

Reply:
Sounds good, we can refactor that in 👍🏻